src/frequenz/quantities/_marshmallow.py

```diff
@@ -0,0 +1,197 @@
+# License: All rights reserved
+# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+
+"""Custom marshmallow fields and schema."""
+
+from typing import Any, Callable
+
+from marshmallow import EXCLUDE, Schema, fields
+
+from ._energy import Energy
+from ._percentage import Percentage
+from ._power import Power
+
+
+class PercentageField(fields.Field):
+    """A field for percentages."""
+
+    def _serialize(
+        self, value: Percentage, attr: str | None, obj: Any, **kwargs: Any
+    ) -> float:
+        """Serialize the percentage to a float.
+
+        Args:
+            value: the percentage
+            attr: the attribute name
+            obj: the object
+            **kwargs: additional keyword arguments
+
+        Returns:
+            The percentage as a float
+        """
+        return value.as_percent()
+
+    def _deserialize(
+        self, value: float, attr: str | None, data: Any, **kwargs: Any
+    ) -> Percentage:
+        """Deserialize the percentage from a float.
+
+        Args:
+            value: the float
+            attr: the attribute name
+            data: the data
+            **kwargs: additional keyword arguments
+
+        Returns:
+            The percentage
+        """
+        return Percentage.from_percent(value)
```

Comment:
Using units here is trickier, because a ratio has no unit, and then if there is no unit is not clear if it is percentage or ratio. That said, I would also parse it as a string, and require the `%` symbol to avoid any ambiguities/confusion (so `-30%` would be valid but `-30` not).